lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -895,6 +901,14 @@ void TFFunctionPartition::diagnoseCopyToAccelerator(
       return;
     }
 
+  auto &ctx = hostFn.getModule().getASTContext();
+
+  // If scalar transfer warnings are turned off, then don't warn about transfers
+  // that are definitely scalars.
+  if (!TFWarnScalarTransfer &&
+      isValidTensorFlowElementType(value->getType().getASTType()))
```

Comment:
It seems when isValidTensorFlowElementType() is given a type Tensor<Float>, it'll extract Float out of it, and return true. Does it really differentiate on scalar shapes?

Say we have `let a = Tensor<Float>([1.0])` or `Tensor<Float>([1.0, 2.0])`, and trigger a send/recv on `a`, would the warning still be on? Would be good to have test coverage on that.